app/src/main/java/com/triviagenai/triviagen/trivia/presentation/TriviaQuestionViewModel.kt

```diff
@@ -29,26 +46,26 @@ class TriviaQuestionViewModel @Inject constructor(
             initialValue = TriviaUIState.Loading
         )
 
-    init {
-        fetchTriviaQuestions("Science")
-    }
-
-    private fun fetchTriviaQuestions(topic: String) {
+    fun fetchTriviaQuestions(topic: String) {
         viewModelScope.launch {
             _uiState.value = TriviaUIState.Loading
             getTriviaQuestionsUseCase(topic).collect { result ->
                 result?.onSuccess { questions ->
-                    _uiState.value = TriviaUIState.Success(questions, 0, 0)
+                    _uiState.value = TriviaUIState.Success(questions.toMutableList(), 0, 0)
                 }?.onFailure { exception ->
                     _uiState.value = TriviaUIState.Error(exception.message ?: "Unknown error")
                 } ?: run { _uiState.value = TriviaUIState.Loading }
             }
         }
     }
 
+    fun fetchRandomTriviaRoundQuestions() {
+        fetchTriviaQuestions(categories.random())
+    }
+
     fun processIntent(intent: TriviaIntent) {
         when (intent) {
-            is TriviaIntent.SubmitAnswer -> submitAnswer(intent.selectedOptionIndex)
+            is TriviaIntent.SubmitAnswer -> viewModelScope.launch { submitAnswer(intent.selectedOptionIndex) }
             TriviaIntent.NextQuestion -> nextQuestion()
```

Comment:
I suppose it was added here when we wanted to make continue button, but now the questions change by themselves, so do we need this NextQuestion? If the NextQuestion is useless, this function can be deleted.